internal/handshake/cipher_suite.go

```diff
@@ -44,12 +48,17 @@ func aeadAESGCMTLS13(key, nonceMask []byte) *xorNonceAEAD {
 	if err != nil {
 		panic(err)
 	}
-	aead, err := cipher.NewGCM(aes)
+	var aead cipher.AEAD
+	if useBoring && boring.Enabled() {
```

Comment:
Should it be an error if `useBoring` but boring isn't enabled? That seems to violate the intention.

Reply:
Sure

Reply:
fixed